File: studentprogressboard/studentprogressboard/Model/Training.cs
using studentprogressboard.Model.Base;
using studentprogressboard.Model.Enums;
using studentprogressboard.Model.Interfaces;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace studentprogressboard.Model
{
    public class Training : Entity, IEntity
    {
        public string Name { get; set; }
        public string? EnglishName { get; set; }
        public string? TrainingCode { get; set; }
        public int NumberOfSemesters { get; set; }
        public string? Language { get; set; }
        public TrainingType TrainingType { get; set; }
        public TrainingMode TrainingMode { get; set; }

        [JsonIgnore]
        [NotMapped]
        public virtual ICollection<Student> Students { get; set; }

        public Training()
        {
            Students = new HashSet<Student>();
        }
    }
}

File: studentprogressboard/studentprogressboard.Client/src/app/_models/training.ts
import { TrainingMode } from "../_enums/training-mode";
import { TrainingType } from "../_enums/training-type";

export class Training{
    id: number;
    name: string;
    englishName: string;
    trainingCode: string;
    numberOfSemesters: number;
    language: string;

    trainingType: TrainingType;
    trainingMode: TrainingMode;
}